parking_permits_app/jwt.py

```diff
@@ -0,0 +1,49 @@
+import jwt
+from django.conf import settings
+from jwt.exceptions import InvalidTokenError
+
+
+class MissingAuthorizationToken(Exception):
+    pass
+
+
+def get_auth_token(request):
+    """
+    Authorization is a bearer token and there for it should have
+    a format of Bearer <TOKEN>
+    """
+    authorization = request.headers.get("Authorization")
+    if not authorization:
+        raise MissingAuthorizationToken("Authorization token is missing.")
+    auth_parts = authorization.split()
+    if len(auth_parts) == 1:
+        raise MissingAuthorizationToken("Authorization token is missing.")
+    return auth_parts[1]
+
+
+def attach_token(func):
+    def wrapper(_, info, *args):
+        token = get_auth_token(request=info.context["request"])
+        decoded = jwt.decode(token, options={"verify_signature": False})
+        user = func(_, info, *args)
+        user.token = jwt.encode(
+            {"id": user.pk, "exp": decoded.get("exp")},
+            settings.JWT_SECRET,
+            algorithm="HS256",
+        )
+        return user
+
+    return wrapper
+
+
+def authenticate_parking_permit_token(func):
+    def wrapper(_, info, *args, **kwargs):
+        token = get_auth_token(request=info.context["request"])
+        try:
+            payload = jwt.decode(token, settings.JWT_SECRET, algorithms=["HS256"])
+        except InvalidTokenError:
+            raise InvalidTokenError("Invalid parking permit authorization token.")
```

Comment:
Must say I'm not familiar with the library, but does jwt automatically process this `InvalidTokenError` and give a 403 response? If that's the case I think we don't need to do try-catch here.

If jwt does not automatically process InvalidTokenError, we could raise a PermissionDenied error here.

Reply:
Yes, It does raise the error. I just try to catch and raise it again to give a proper message as we are handling two auth token in our case.

Reply:
OK, I thnk we can go with it